keps/sig-scheduling/4671-gang-scheduling/README.md

```diff
@@ -0,0 +1,983 @@
+- scheduler changes implementing the gang scheduling functionality
+
+When user upgrades the cluster to the version that supports these two features:
+- they can start using the new API by creating Workload objects and linking pods to it via
+  explicitly specifying their new `spec.workload` field
+- scheduler automatically uses the new extensions and tries to schedule all pods from a given
+  gang in a scheduling group based on the defined `Workload` objects
+
+When user downgrades the cluster to the version that no longer supports these two features:
+- the `Workload` objects can no longer be created (the existing ones are not removed though)
+- the `spec.workload` field can no longer be set on the Pods (the already set fields continue
+  to be set though)
+- scheduler reverts to the original behavior of scheduling one pod at a time ignoring
+  existence of `Workload` objects and pods being linked to them
+
+
+### Version Skew Strategy
+
+The feature is limited to the control plane, so the version skew with nodes (kubelets) doesn't matter.
+
+For the API changes (introduction of Workload API and the new field in Pod API), the old version of
+components (in particular kube-apiserver) may not handle those. Thus, users should not set those
+fields before confirming all control-plane instances were upgraded to the version supporting those.
+
+For the gang-scheduling itself, this is purely kube-scheduler in-memory feature, so the skew doesn't
+really matter (as there is always only single kube-scheduler instance being a leader).
+
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: Workload/GenericWorkload/NativeWorkload
```

Comment:
`kep.yaml` talks about `Workload`, would be nice to pick one and use it consistently across all places :wink: